src/optimizer/stats_calculator.cpp

```diff
@@ -64,18 +64,19 @@ void StatsCalculator::Visit(const LogicalGet *op) {
         AddBaseTableStats(col, table_stats, predicate_stats, false);
       }
     }
-    // Use predicates to update the stats accordingly
-    UpdateStatsForFilter(
-        table_stats->GetColumnCount() == 0 ? 0 : table_stats->num_rows,
-        predicate_stats, op->predicates);
+    // Use predicates to estimate cardinality
+    if (table_stats->GetColumnCount() == 0) {
+      root_group->SetNumRows(0);
```

Comment:
This would happen if [`ColumnStatsCatalog::GetTableStats`](https://github.com/cmu-db/peloton/blob/e738acb956306d6914a5a94e6773f410a3eae4a1/src/include/catalog/column_stats_catalog.h#L86-L90) returned no column stats. I believe this would happen if the catalog did not have stats for the table. This is assuming its not possible to have a table without columns.